### ISSUE TITLE
[Masthead] Resolve a11y issues identified by storybook in web components - DDS Consulting

### ISSUE THREAD

**Issue by jeffchew** [carbon-design-system/carbon-for-ibm-dotcom#7128]
Describe the issue: 

> What did you expect to happen? What happened instead? 

Address all accessibility violations identified in storybook:

https://www.ibm.com/standards/carbon/web-components/?path=/story/components-masthead--default

![image](https://user-images.githubusercontent.com/1641214/133140387-80513950-bc95-4078-8b7f-9d32cf864e72.png)

> Component(s) impacted? Please include screenshots, or code and relevant URLs.

Masthead (web components)

## Additional information
-  Engineering info:  
     - HC engineer: @huntermacd 
     - JC Jira Ticket: https://jsw.ibm.com/browse/HC-2202
     - DDS consulting engineer:  @kennylam
